Broker reconciler tests create EventPolicies without enabling required OIDC feature flags

### Description

Some broker reconciler tests create and depend on `EventPolicy` resources without enabling the required `oidc-authentication` feature flag in the test context.

With admission validation enabled, `EventPolicy` creation is rejected when OIDC authentication is disabled. The resource is therefore never created, but the test framework continues waiting for it and eventually times out. This results in failures that are difficult to diagnose and obscure the real cause.

---

### Failing Test output :

```text
--- FAIL: TestBrokerSupportsAuthZ/Broker_handles_authorization_features_correctly/Broker_only_admits_events_that_pass_the_event_policy_filter_with_binary_encoding_for_input_event/Setup/Install_the_EventPolicy (2.07s)
FAIL test/e2e_new.TestBrokerSupportsAuthZ/Broker_handles_authorization_features_correctly/Broker_only_admits_events_that_pass_the_event_policy_filter_with_binary_encoding_for_input_event/Setup/Install_the_EventPolicy (2.07s)

eventpolicy.go:53: failed to create resource admission webhook "validation.webhook.eventing.knative.dev" denied the request:
validation failed: oidc-authentication feature not enabled
  - Resource:
      apiVersion: eventing.knative.dev/v1alpha1
      kind: EventPolicy
      metadata:
        name: eventpolicy-gneswxqh

wait.go:136: test-ghizcwkz eventpolicy-gneswxqh not found eventpolicies.eventing.knative.dev
steps.go:47: eventing.knative.dev/v1alpha1, Resource=eventpolicies did not become ready, timed out waiting for the condition

--- FAIL: TestKafkaSinkSupportsAuthZ/KafkaSink_handles_authorization_features_correctly/KafkaSink_only_admits_events_that_pass_the_event_policy_filter_with_binary_encoding_for_input_event/Setup/Install_the_EventPolicy (0.39s)

eventpolicy.go:53: failed to create resource admission webhook "validation.webhook.eventing.knative.dev" denied the request:
validation failed: oidc-authentication feature not enabled
  - Resource:
      apiVersion: eventing.knative.dev/v1alpha1
      kind: EventPolicy
      metadata:
        name: eventpolicy-lbcfuxtw

wait.go:136: test-kobgrpjp eventpolicy-lbcfuxtw not found eventpolicies.eventing.knative.dev
steps.go:47: eventing.knative.dev/v1alpha1, Resource=eventpolicies did not become ready, timed out waiting for the condition

-----------------------------------------------------------------------------------------------------------
### Proposed Change

For tests that create or evaluate EventPolicy resources,
 explicitly enable the required feature flags in the test context (for example, enabling OIDC-authentication and an appropriate authorization mode) so that test behavior aligns with admission validation.

--- a/test/reconciler-tests.sh
+++ b/test/reconciler-tests.sh
@@ -76,28 +76,27 @@ echo "Running E2E Reconciler OIDC and AuthZ Tests"
 
 kubectl apply -Rf "$(dirname "$0")/config-auth"
 
-# Wait until we can actually create an EventPolicy (proof webhook has the right config)
-echo "Waiting for webhook to reload OIDC config..."
+# Wait until ALL webhook pods have reloaded the OIDC config
+echo "Waiting for all webhook pods to reload OIDC config..."
 for i in {1..60}; do
-  if kubectl create -f - <<EOF 2>/dev/null
-apiVersion: eventing.knative.dev/v1alpha1
-kind: EventPolicy
-metadata:
-  name: config-readiness-test
-  namespace: knative-eventing
-spec:
-  to:
-    - ref:
-        apiVersion: v1
-        kind: Service
-        name: test
-EOF
-  then
-    kubectl delete eventpolicy config-readiness-test -n knative-eventing
-    echo "Webhook ready with OIDC config enabled"
+  # Get all eventing-webhook pod names
+  webhook_pods=$(kubectl get pods -n knative-eventing -l app=eventing-webhook -o jsonpath='{.items[*].metadata.name}')
+
+  all_ready=true
+  for pod in $webhook_pods; do
+    # Check if this pod has logged the authentication-oidc: enabled config
+    if ! kubectl logs -n knative-eventing "$pod" --tail=100 2>/dev/null | grep -q 'authentication-oidc:Enabled'; then
+      echo "Attempt $i: Pod $pod not ready yet..."
+      all_ready=false
+      break
+    fi
+  done
+
+  if [ "$all_ready" = true ]; then
+    echo "All webhook pods ready with OIDC config enabled"
     break
   fi
-  echo "Attempt $i: Webhook not ready yet, waiting..."
+
   sleep 2
 done
 

